XamarinCommunityToolkit/Behaviors/MatchTypes/MentionMatchType.shared.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Xamarin.Forms;
+namespace Microsoft.Toolkit.Xamarin.Forms.Behaviors
+{
+    public class MentionMatchType : MatchType
+    {
+        public override string Regex => @"[@]\w+";
```

Comment:
Since it is a readonly property and should be a regex pattern, why not use `Regex` type?
We can use the [Compiled Regex](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices) to improve some performance when re-use it.
It will cost some initilialize time, so maybe we can do a lazy start for it...


```suggestion
        public override Lazy<Regex> Regex => new Lazy<Regex>(()=> new Regex(@"[@]\w+", RegexOptions.Compiled | RegexOptions.Singleline));
```

Reply:
Ive created a simple test on dotnetFiddle, so the results are:
https://dotnetfiddle.net/d76YZs

![image](https://user-images.githubusercontent.com/19656249/89669261-84fbdc00-d8b5-11ea-9173-806732900534.png)

Reply:
Ive created a simple test on dotnetFiddle, so the results are:
https://dotnetfiddle.net/d76YZs

![image](https://user-images.githubusercontent.com/19656249/89669261-84fbdc00-d8b5-11ea-9173-806732900534.png)

Reply:
I guess it should be 
```csharp
public override Lazy<Regex> Regex { get; } = ...;
```
instead of
```csharp
public override Lazy<Regex> Regex => ...;
```

Reply:
Yeah! my bad!